### DISCUSSION TITLE
How can I prevent the name of the fields in a pivot table from having an auto-generated name?

### DISCUSSION THREAD

**Discussion by sawa-ko** [mikro-orm/mikro-orm#4671] [Q&A] [answered]
Currently I have a pivot table, generated by mikro-orm, but I see that the name of the columns has a format that I don't want, autogenerated, but I want them to have as name: role_id and member_id, for example.

![image](https://github.com/mikro-orm/mikro-orm/assets/56084970/b06a4764-2eec-4b4e-82d9-74f3e2742d4d)

# Entities

```ts
export class MemberEntity {
  @ManyToMany(() => BoardRolesEntity, (role) => role.members, {
    comment: 'User member roles in the board.',
    owner: true,
  })
  public roles = new Collection<BoardRolesEntity>(this);
}

export class RolEntity {
  @ManyToMany(() => MemberEntity , (member) => member.roles, {
    comment: 'Board members who have this role.',
  })
  public members = new Collection<MemberEntity >(this);
}
```

**Comment by sawa-ko** [comment]
Can you help me please @B4nan.

**Comment by B4nan** [comment, marked as answer]
There are several ways:
- m:n decorator options, namely `joinColumn` and `inverseJoinColumn` (https://mikro-orm.io/docs/decorators#manytomany)
- defining your own pivot entity(https://mikro-orm.io/docs/collections#custom-pivot-table-entity)
- overriding the naming strategy (https://mikro-orm.io/docs/naming-strategy)

**Reply by sawa-ko** [reply]
Not working with `joinColumn` and `inverseJoinColumn`.

> Result

![image](https://github.com/mikro-orm/mikro-orm/assets/56084970/d635234f-4922-44e5-94f7-3a48c301fad6)

> Code

```ts
  @ManyToMany(() => ProjectRolesEntity, (r) => r.members, {
    comment: 'User member roles in the project.',
    joinColumn: 'role_id',
    owner: true,
  })
  public roles = new Collection<ProjectRolesEntity>(this);

  @ManyToMany(() => ProjectMembersEntity, (m) => m.roles, {
    comment: 'Project members who have this role.',
    inverseJoinColumn: 'member_id',
  })
  public members = new Collection<ProjectMembersEntity>(this);
```

Am I doing something wrong? @B4nan

**Reply by B4nan** [reply]
You need to set both of those flags on the owning side.
